recipes/finetuning/README.md

```diff
@@ -25,30 +25,35 @@ It lets us specify the training settings for everything from `model_name` to `da
 ```python
 
 model_name: str="PATH/to/LLAMA 2/7B"
-enable_fsdp: bool= False
+enable_fsdp: bool=False
 run_validation: bool=True
 batch_size_training: int=4
 gradient_accumulation_steps: int=1
+max_train_step: int=0
+max_eval_step: int=0
 num_epochs: int=3
 num_workers_dataloader: int=2
 lr: float=2e-4
 weight_decay: float=0.0
-gamma: float= 0.85
+gamma: float=0.85
 use_fp16: bool=False
 mixed_precision: bool=True
 val_batch_size: int=4
 dataset = "samsum_dataset" # alpaca_dataset, grammar_dataset
-peft_method: str = "lora" # None , llama_adapter, prefix
+peft_method: str="lora" # None , llama_adapter, prefix
 use_peft: bool=False
-output_dir: str = "./ft-output"
+output_dir: str="./ft-output"
 freeze_layers: bool = False
 num_freeze_layers: int = 1
 quantization: bool = False
 save_model: bool = False
 dist_checkpoint_root_folder: str="model_checkpoints"
 dist_checkpoint_folder: str="fine-tuned"
 save_optimizer: bool=False
-
+flop_counter: bool=False # Enable FLOPS counter to measure model throughput, can not be used with pytorch profiler at the same time.
+flop_counter_start: int=3 # The step to start profiling, default is 3, which means after 3 steps of warm-up stage, the profiler will start to count FLOPS.
+use_profiler: bool=False # Enable pytorch profiler, can not be used with FLOPS counter at the same time.
+profiler_dir: str="PATH/to/save/profiler/results" # will be used if using profiler
```

Comment:
I was following the current example of [config.py](https://github.com/meta-llama/llama-recipes/blob/main/src/llama_recipes/configs/training.py), where we have both --enable_fsdp and --dist_checkpoint_root_folder. However, @HamidShojanazeri @mreso  I noticed that there is a discrepancy between the [readme](https://github.com/meta-llama/llama-recipes/tree/main/recipes/finetuning#how-to-configure-finetuning-settings) and the actual [training config](https://github.com/meta-llama/llama-recipes/blob/main/src/llama_recipes/configs/training.py#L40). For example, dist_checkpoint_root_folder: str="model_checkpoints" vs dist_checkpoint_root_folder: str="PATH/to/save/FSDP/model".  Should I changed the readme according to the configs/training.py?

Reply:
Would be good to align the readme or create a followup PR for this. @HamidShojanazeri any preference regarding the --enable_X pattern vs using the dir parameter directly. In case we go with the two parameter solution we should align with the fsdp parameter and use --enable_profiler.

Reply:
Also if choose to use -- output_dir, should we predefine a folder name or still something like /Path/to/save/..., eg  dist_checkpoint_root_folder: str="model_checkpoints" vs dist_checkpoint_root_folder: str="PATH/to/save/FSDP/model". The readme and training config now have two different format..

Reply:
@HamidShojanazeri  Do you have any preference about this?

Reply:
thanks @mreso for the review lets keep it consistent with our previous design for now then we can revisit.